lib/Serialization/Deserialization.cpp

```diff
@@ -4555,6 +4504,93 @@ void ModuleFile::loadAllMembers(Decl *container, uint64_t contextData) {
   }
 }
 
+static Decl* handleErrorAndFudgeMissingMember     (llvm::Error &&,  ASTContext &,  Decl*         container);
+static Decl* handleErrorAndFudgeMissingClassMember(llvm::Error &&,  ASTContext &,  ClassDecl*    container);
+static Decl* handleErrorAndFudgeMissingProtoMember(llvm::Error &&,  ASTContext &,  ProtocolDecl* container);
+static Decl* handleErrorAndFudgeMissingMiscMember (llvm::Error &&);
+
+Decl* ModuleFile::getOrFudgeOneMember(Decl *container, DeclID rawID) {
+  Expected<Decl *> next = getDeclChecked(rawID);
+  
+  if (next) {
+    assert(next.get() && "unchecked error deserializing next member");
+    return next.get();
+  }
+  llvm::Error &&error = next.takeError();
+  if (!getContext().LangOpts.EnableDeserializationRecovery)
+    fatal(std::move(error));
+
+  return handleErrorAndFudgeMissingMember(std::move(error), getContext(), container);
+}
+
+
+Decl* handleErrorAndFudgeMissingMember(llvm::Error &&error, ASTContext& context, Decl *container) {
+  auto *containingClass = dyn_cast<   ClassDecl>(container);
+  auto *containingProto = dyn_cast<ProtocolDecl>(container);
+  
+  // Drop the member if it had a problem.
+  // FIXME: Handle overridable members in class extensions too, someday.
+
+  return
+  containingClass ?  handleErrorAndFudgeMissingClassMember(std::move(error), context, containingClass)  :
+  containingProto ?  handleErrorAndFudgeMissingProtoMember(std::move(error), context, containingProto)  :
+                     handleErrorAndFudgeMissingMiscMember (std::move(error));
+}
+
+Decl* handleErrorAndFudgeMissingClassMember(llvm::Error &&error, ASTContext &context, ClassDecl* containingClass) {
+  Decl* classMember = nullptr;
+  auto handleErrorAndFudgeMissingClassMember =
```

Comment:
I don't think it's adding much to name the closure before passing it here; IOW you can save a bit of chatter and write:

Decl* classMember = nullptr;
llvm::handleAllErrors(std::move(error),
    [&](const DeclDeserializationError &error) {
  ...
});
return classMember;